abel/linbasex.py

```diff
@@ -0,0 +1,418 @@
+
+import numpy as np
+import abel
+import scipy as sci
+from scipy.special import eval_legendre
+from scipy import ndimage
+
+###############################################################################
+# linbasex - inversion procedure based on 1-dimensional projections of
+#            velocity-map images
+#
+# As described in:
+#   Gerber, Thomas, Yuzhu Liu, Gregor Knopp, Patrick Hemberger, Andras Bodi,
+#   Peter Radi, and Yaroslav Sych,
+#     “Charged Particle Velocity Map Image Reconstruction with One-Dimensional
+#      Projections of Spherical Functions.”
+#     Review of Scientific Instruments 84, no. 3 (March 1, 2013):
+#                                      033101–033101 – 10.
+#     doi:10.1063/1.4793404.
+#
+# 2016-04- Thomas Gerber and Daniel Hickstein - theory and code updates
+# 2016-04- Stephen Gibson core code extracted from the supplied jupyter
+#          notebook (see #167: https://github.com/PyAbel/PyAbel/issues/167)
+#
+###############################################################################
+
+_linbasex_parameter_docstring = \
+    r"""Inverse Abel transform using 1d projections of VM-images.
+
+    `Gerber, Thomas, Yuzhu Liu, Gregor Knopp, Patrick Hemberger, Andras Bodi,
+    Peter Radi, and Yaroslav Sych.
+    Charged Particle Velocity Map Image Reconstruction with One-Dimensional Projections of Spherical Functions.` Review of Scientific Instruments 84, no. 3 (March 1, 2013): 033101–033101 – 10.
+    <http://dx.doi.org/10.1063/1.4793404>`_
+
+     linbasex determines the contribution of spherical functions,
+     calculating the weight of each :math:`Y_{i0}`. The reconstructed 3D object
+     is obtained by adding all the :math:`Y_{i0}` contributions, from which
+     slices are derived.
+
+
+    Parameters
+    ----------
+    Dat: numpy 2D array
+        image data must be square shape of odd size
+    an: list
+        projection angle, in degrees
+        e.g. [0, 90] or [0, 54.7356, 90] or [0, 45, 90, 135]
+    un: list
+        order of Legendre polynomials to be used as the expansion
+            even polynomials [0, 2, ...] gerade
+            odd polynomials [1, 3, ...] ungerade
+            all orders [0, 1, 2, ...].
+        In a single photon experiment there are only anisotropies up to
+        second order. The interaction of 4 photons (four wave mixing) yields
+        anisotropies up to order 8.
+    inc: int
+        number of pixels per Newton sphere (default 1)
+    sig_s: float
+        sigma for smoothing (default 0.5)
+    rcond: float
+        (default 0.0005) scipy.linalg.lstsq fit conditioning value.
+        set rcond to zero to switch conditioning off
+    threshold: float
+        threshold for normalization of higher order Newton spheres
+        (default 0.2)
+    clip: int
+        clip first vectors (smallest Newton spheres) to avoid singularities
+    norm_range: tuple 
+        (low, high)
+        normalization of Newton sphere, maximum in range
```

Comment:
I guess the above description is well done. Maybe one could point out: There is no clue to consider "anisotropies" of contributions hardly above noise. Therefore choose threshold at least above noise.

Indeed, I chose a quite arbitrary way to define "norms".
The easiest thing would be just to think in terms of counts/sphere. We note that Beta0[i] produces the total number of counts integrated over sphere i. We also note that the integral number of counts in every non isotropic sphere is zero. (The integral over a Legendre polynomial is zero with exception of P_0). If we describe an anisotropic distribution of charged particles in a VMI we can describe it as an addition of an isotropic part (Beta0[i]) containing the total number of counts. Adding "anisotropic" spheres just add the same amount of counts at certain angles that are subtracted at other angles.